README.md

```diff
@@ -1,41 +1,108 @@
+
+```
+var isGranted = await locationService.RequestPermission(ct);
+```
+
+You may safely request the permission multiple times; the user will only get prompted as often as is necessary.
+
+For instance, if you ask the user for their permission once and they decline, calling ```RequestPermission``` again will not prompt the user again, but simply return ```false```.
+
+If the user answers "Allow once", then calling ```RequestPermission``` again during the lifetime of the app will simply return ```true```. Once the app is restarted, calling ```RequestPermission``` will prompt them for the permission once again, since their original permission has now expired.
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+Once permission is requested, you can use the various methods and extension methods of IGeolocatorService.
 
 ## Features
 
-{More details/listing of features of the project}
+### Request the user's location
+
+Once you have obtained the location permission, it's a simple matter to obtain the location:
+
+```
+var location = await locationService.GetLocation(ct);
+```
+
+If the user has denied the location permission, an ```InvalidOperationException``` is thrown; therefore, we recommend not calling this method if you know the user has denied the location permission and this is a normal scenario. If the app has not kept track of this information, we recommend calling this extension method instead:
+
+```
+var locationOrNull = await locationService.GetLocationOrNull(ct);
+```
+
+This method will return null instead of throwing an exception, in case the permission was denied.
+
+### React to changes in the user's location or their permissions
+
+```IGeolocatorService``` offers events to allow you to react when the user's location has changed, or when they have granted or denied permissions. Even if these changes occur while the app is in background, they will be raised when the app comes back to the foreground.
+
+```
+locationService.LocationChanged -= RefreshUserLocation;
+locationService.LocationChanged += RefreshUserLocation;
+
+locationService.LocationPermissionChanged -= OnPermissionChanged;
+locationService.LocationPermissionChanged += OnPermissionChanged;
+```
+
+### Track the location and permission using Reactive Extensions
+
+GeolocatorService.Reactive adds a few extension methods which will allow your app to fluidly keep track of the user's location and permission status.
+
+For instance, let's say your app wants to display relevant information to the user's location when it's available, and a completely different type of information when that location is unavailable (because the permission was denied or the GPS is not working).
+
```

Comment:
Location services use cellular data, wifi location, bluetooth and gps. I suggest `because the permission was denied or the phone is unable to provide a location).`